### PR TITLE
feat: add base cartridge and setup scripts

### PR DESCRIPTION
**Pull request by luqven**
This PR adds a base imgix cartridge for testing purposes. It also adds useful packages and scripts for cartridge and sandbox management.

This PR should:
- allow you to upload and make changes to the `app_imgix_base_sfra` cartridge in a sandbox
- allow you to use npm or yarn to run QOL sandbox maintenance scripts, like `yarn sandbox:list`
- allow you to us the Prophet VSCode extension to manage cartridges

## Available Scripts

Bellow are some of the scripts added to this PR. There's some setup involved in order to use them.

### Store the SFCC CLI Credentials

To manage your sandboxes, it’s easiest to use the `sfcc-ci`  package. You can `npm i -G sfcc-ci` to use it anywhere, or `npm i sfcc-ci` to use it locally.

The package expects certain ENV variables to be available. You can add them to your bash profile or into a `.env` file.

``` bash
# SFCC CLI
SFCC_OAUTH_CLIENT_ID='<SFCC_OAUTH_CLIENT_ID>'
SFCC_OAUTH_CLIENT_SECRET='<SFCC_OAUTH_CLIENT_SECRET>'
SFCC_OAUTH_USER_NAME='<SFCC_OAUTH_USER_NAME>'
SFCC_OAUTH_USER_PASSWORD='<SFCC_OAUTH_USER_PASSWORD>'
SFCC_DEV_SANDBOX_ID='<SFCC_DEV_SANDBOX_ID>'
# [Optional] DW JSON, I store these as well to manage Prophet with ENV variables
SFCC_HOSTNAME='<SFCC_HOSTNAME>'
SFCC_USERNAME=$SFCC_OAUTH_USER_NAME
SFCC_PASSWORD=$SFCC_OAUTH_USER_PASSWORD
SFCC_CODE_VERSION='<SFCC_CODE_VERSION>'
```

### Manage sandboxes
Once you have `sfcc-ci` installed and your ENV variables set, you can use it to manage sandboxes, like so:

``` bash
sfcc-ci client:auth    # login to the sfcc-ci client
sfcc-ci sandbox:list   # list all available sandboxes
sfcc-ci sandbox:start  # start the sandbox, IE $SFCC_DEV_SANDBOX_ID
sfcc-ci sandbox:stop   # stop the sandbox
```

### Store the DW credentials

Next, you need to create a `dw.json` file at the project root that stores your SFCC credentials. The Prophet extension will use this file to read and write to and from your SFCC sandbox. Also, the `sgmf` upload scripts use it to upload to your sandbox.

You can either do this manually or run `yarn setup` to have the `setup.js` script create it for you.

```js
    // dw.json
    {
        "hostname": "your-sandbox-hostname.demandware.net",
        "username": "yourlogin",
        "password": "yourpwd",
        "code-version": "version_to_upload_to"
    }
```

### CODE REVIEW

**Review by frederickfogerty — CHANGES_REQUESTED** [review]
This looks great @luqven. Just a small thing, before merging, could you please rename the "Adding SFRA Extention cartridge with page designer test file" to fix the typo and make it conform to commitlint conventions?

Also, would you mind adding an explanation for the cartridge `app_imgix_base_sfra` in Paper [here](https://paper.dropbox.com/doc/commerce-cloud-MVP-Technical-Planning--BS_3rQLZ2CB~i7cHCyasSwKHAg-Yi2EKxcbu7ZG57c7UzEmj)? Thanks 🙌

File: cartridges/app_imgix_base_sfra/cartridge/experience/components/imgix/testcomp.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+var HashMap = require('dw/util/HashMap');
+// we included an existing SFCC back-end script to handle the image path
+var ImageTransformation = require('*/cartridge/experience/utilities/ImageTransformation.js');
+
+// handle function for component.
+module.exports.render = function (context, modelIn) {
+    var model = modelIn || new HashMap();
+    var content = context.content;
+
+    model.title = content.title ? content.title : null;
```
Comment (luqven) [inline cartridges/app_imgix_base_sfra/cartridge/experience/components/imgix/testcomp.js:13]: Reason `model.title` will be `null` is that it doesn't match the `.json` spelling `tittle`.

File: cartridges/app_imgix_base_sfra/cartridge/experience/components/imgix/testcomp.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    model.image = ImageTransformation.getScaledImage(content.image);
+    // use to give link on the image, if we click on image it take to us to that page.
+    model.link = content.ITCLink ? content.imageLink : '#';
```
Comment (luqven) [inline cartridges/app_imgix_base_sfra/cartridge/experience/components/imgix/testcomp.js:16]:
```suggestion
    model.link = content.imageLink ? content.imageLink : '#';
```

File: package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "sf-commerce-cloud",
+  "version": "0.0.1",
+  "description": "Use this integration to insert images from imgix's Image Manager into your Salesforce Commerce Cloud websites",
+  "main": "index.js",
+  "repository": "https://github.com/imgix/sf-commerce-cloud.git",
+  "author": "sdk@imgix.com",
+  "license": "MIT",
```
Comment (frederickfogerty) [inline package.json:8]: Let's update this license 👍

**Review by frederickfogerty — COMMENTED** [review]
LGTM ✅

**Pull request merged**